cmd/token_test.go

```diff
@@ -89,6 +89,43 @@ func TestTokenWithGCEAK(t *testing.T) {
 	}
 }
 
+func TestCustomEventLogFile(t *testing.T) {
+	ExternalTPM = nil
+	var dummyMetaInstance = util.Instance{ProjectID: "test-project", ProjectNumber: "1922337278274", Zone: "us-central-1a", InstanceID: "12345678", InstanceName: "default"}
+	mockMdsServer, err := util.NewMetadataServer(dummyMetaInstance)
+	if err != nil {
+		t.Error(err)
+	}
+	defer mockMdsServer.Stop()
+
+	mockOauth2Server, err := util.NewMockOauth2Server()
+	if err != nil {
+		t.Error(err)
+	}
+	defer mockOauth2Server.Stop()
+
+	// Endpoint is Google's OAuth 2.0 default endpoint. Change to mock server.
+	google.Endpoint = oauth2.Endpoint{
+		AuthURL:   mockOauth2Server.Server.URL + "/o/oauth2/auth",
+		TokenURL:  mockOauth2Server.Server.URL + "/token",
+		AuthStyle: oauth2.AuthStyleInParams,
+	}
+
+	mockAttestationServer, err := util.NewMockAttestationServer()
+	if err != nil {
+		t.Error(err)
+	}
+	defer mockAttestationServer.Stop()
+
+	RootCmd.SetArgs([]string{"token", "--verifier-endpoint", mockAttestationServer.Server.URL, "--event-log", "/test-event-log"})
+	if err := RootCmd.Execute(); err != nil {
+		if err.Error() != "failed to attest: failed to retrieve TCG Event Log: open /test-event-log: no such file or directory" &&
+			err.Error() != "connecting to TPM: stat /dev/tpm0: no such file or directory" {
```

Comment:
Why does this check two error strings? Is the failure not deterministic?

Reply:
Also my question. Seems like the test will always pass (at least in a CICD) because the /dev/tpm0 not found. So the actual eventlog behavior is not tested.

The behavior is depending on the test environment which is not ideal.

Reply:
I quoted my reply above "This test tests the eventlog flag. And this test is expected to open /dev/tpm0 and /test-event-log. So this test is expected to return failed to open /dev/tpm0 error for environments with no TPM and failed to open /test-event-log for environments with a TPM."

Reply:
We could also make this test to skip for for environments with no TPM. What do you think?

Reply:
yes I think skipping is better

Reply:
Got it, thanks!